bleps-macros/src/lib.rs

```diff
@@ -241,32 +241,32 @@ pub fn gatt(input: TokenStream) -> TokenStream {
                                                             }
                                                         }
                                                     } else {
-                                                        return quote!{ compile_error!("Unexpected"); }.into();
+                                                        return quote!{ compile_error!("Characteristic field 'descriptors' must be an array"); }.into();
                                                     }
                                                 }
                                                 _ => {
-                                                    return quote! { compile_error!("Unexpected"); }
+                                                    return quote! { compile_error!("Unexpected characteristic field '{}'", name); }
                                                         .into()
                                                 }
                                             }
                                         }
 
                                         service.characteristics.push(charact);
                                     } else {
-                                        return quote! { compile_error!("Unexpected"); }.into();
+                                        return quote! { compile_error!("Characteristic definition must be given as 'characteristic { ... }'"); }.into();
                                     }
                                 }
                             } else {
-                                return quote! { compile_error!("Unexpected"); }.into();
+                                return quote! { compile_error!("Service field 'characteristics' must be an array"); }.into();
                             }
                         }
-                        _ => return quote! { compile_error!("Unexpected"); }.into(),
+                        _ => return quote! { compile_error!("Unexpected service field '{}'", name); }.into(),
```

Comment:
This will just print `Unexpected service field '{}'` - you can make it print the right thing with something like `quote! { compile_error!(concat!("Unexpected service field ", #name)); }.into()`